### ISSUE TITLE
Issue with layout on reload with stateSave: true

### ISSUE THREAD

**Issue by mbalslow** [DataTables/SearchBuilder#2]
I experience that with stateSave: true, it correctly saves the filters and on reload, it display them, but the layout seems off.

My settings are simply:
```
{
  "dom": "Qlfrtip",
  "stateSave": true
}
```

I have the CSS files in the header:
```
<link rel="stylesheet" href="https://cdn.datatables.net/1.10.22/css/jquery.dataTables.css">
<link rel="stylesheet" href="https://cdn.datatables.net/searchbuilder/1.0.1/css/searchBuilder.dataTables.min.css">
```

And script tags in my body:
```
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"</script>
<script src="https://cdn.datatables.net/1.10.22/js/jquery.dataTables.min.js"></script>
<script src="https://cdn.datatables.net/searchbuilder/1.0.1/js/dataTables.searchBuilder.min.js"></script>
```

Here is a screenshot of how it looks before refreshing the page:

![image](https://user-images.githubusercontent.com/49169557/102939930-794d3280-44af-11eb-8fe8-d640cac844fd.png)

After refreshing the page, this is how it looks:

![image](https://user-images.githubusercontent.com/49169557/102939966-89651200-44af-11eb-8ca8-acb277e310e1.png)

By simply resizing the browser window, the searchBuilder resizes and the layout looks correct again.

**Comment by mbalslow** [comment]
It seems to be an issue with the positioning of the "dtsb-logicContainer". It is placed too far to the left on the initial reload.

![image](https://user-images.githubusercontent.com/49169557/102941271-933c4480-44b2-11eb-8318-f4a506c68979.png)

As soon as I resize the browser window even slightly, the elements are repositioned to be in the correct spot.

![image](https://user-images.githubusercontent.com/49169557/102941360-d0083b80-44b2-11eb-9136-c93a4b333077.png)

**Comment by SandyDatatables** [comment]
Hi @mbalslow ,

Thanks for your question. Please in future could you direct support requests to the [DataTables Forums](https://datatables.net/forums/). This is our preferred way of managing such requests, and allows the post to be reached more easily by other members of the community.

This isn't behaviour that I am seeing locally. Please could you  post a link to a running test case showing the issue so we can offer some help. Information on how to create a test case (if you aren't able to link to the page you are working on) is [available here](https://datatables.net/manual/tech-notes/10).

Thanks,
Sandy

**Comment by mbalslow** [comment]
Hi @SandyDatatables,

Thanks. Should I go ahead and post the issue in the forums now, or make sure to do it going forward? I will do whatever is easiest for you!

I have made a live test page here that illustrates the issue (stripped-down version of my actual page): https://jsfiddle.net/yuv76wjL/1/

To recreate the issue, open the jsFiddle, add one or two conditions and reload the page.

In the DOM, I have an empty table. In the real case, I perform several HTTP requests to gather the data. In the test case, I've just hardcoded some random data.

To improve the user experience, I show a loader and have "display: none" on the table while the data is retrieved. When the data is retrieved, I hide the loader and display the table. For the test case, I have just hardcoded a fake wait of 1 second.

My function to activate the dataTable starts at line 58.

I believe that the issue is in fact created by the fact that I initiate my datatable with "display: none". If I don't do that, the issue will not occur.

I hope this provides you with sufficient information, otherwise, please let me know.

Thanks!
Mathias

**Comment by SandyDatatables** [comment]
Hi @mbalslow ,

No worries, we can just sort this here this time around :slightly_smiling_face: 

Looks as though it is the display none that is causing the issues. Editing your example shows [it working](https://jsfiddle.net/8hbyv7ko/1/).

SearchBuilder determines the positioning of all of the elements that is uses dynamically, so it needs to be visible for this to work. 

A fix here however is to make use of the API methods [searchBuilder.rebuild()](https://datatables.net/reference/api/searchBuilder.rebuild()) and [searchBuilder.getDetails() ](https://datatables.net/reference/api/searchBuilder.getDetails())to reconstruct SearchBuilder once the table has loaded. Take a look at this version of [your example](https://jsfiddle.net/8hbyv7ko/4/).

Hope this helps,
Sandy

**Comment by mbalslow** [comment]
Hi again @SandyDatatables,

Amazing, that makes perfect sense. I actually already attempted to solve the issue using this approach, however, at that moment, I hadn't realized that the issue is caused by the display none. So I did the rebuilding right before making the table visible, not right after, which now in hindsight doesn't make much sense.

But anyway, thank you for the quick support - and for an awesome tool 💪
